testworld/funding_test.go

```diff
@@ -0,0 +1,150 @@
+		if status != "success" {
+			t.Error(message)
+		}
+
+		fundingId := getFundingId(t, res)
+		fundings = append(fundings, fundingId)
+
+	}
+
+	params := map[string]interface{}{
+		"document_id": docIdentifier,
+		"currency":    "USD",
+		"amount":      "20000",
+		"apr":         "0.33",
+	}
+
+	getListFundingCheck(alice.httpExpect, alice.id.String(), docIdentifier, 6, params)
+	getListFundingCheck(bob.httpExpect, bob.id.String(), docIdentifier, 6, params)
+	getListFundingCheck(charlie.httpExpect, charlie.id.String(), docIdentifier, 6, params)
+
+}
+
+func signTest(t *testing.T, alice, bob, charlie hostTestSuite, fundingId, docIdentifier string) {
+	// alice adds a funding and shares with charlie
+	res := signFunding(alice.httpExpect, alice.id.String(), docIdentifier, fundingId, http.StatusOK)
+	txID := getTransactionID(t, res)
+	status, message := getTransactionStatusAndMessage(alice.httpExpect, alice.id.String(), txID)
+	if status != "success" {
+		t.Error(message)
+	}
+
+	fundingId = getFundingId(t, res)
+	params := map[string]interface{}{
+		"document_id": docIdentifier,
+		"currency":    "USD",
+		"amount":      "20000",
+		"apr":         "0.33",
+	}
+
+	// check if everybody received to funding with a signature
+	getFundingWithSignatureAndCheck(alice.httpExpect, alice.id.String(), docIdentifier, fundingId, "true", "false", params)
+	getFundingWithSignatureAndCheck(bob.httpExpect, bob.id.String(), docIdentifier, fundingId, "true", "false", params)
+	getFundingWithSignatureAndCheck(charlie.httpExpect, charlie.id.String(), docIdentifier, fundingId, "true", "false", params)
+
+}
+
+func createInvoiceWithFunding(t *testing.T, alice, bob, charlie hostTestSuite) (fundingId, docIdentifier string) {
+	res := createDocument(alice.httpExpect, alice.id.String(), typeInvoice, http.StatusOK, defaultInvoicePayload([]string{bob.id.String()}))
+	txID := getTransactionID(t, res)
+	status, message := getTransactionStatusAndMessage(alice.httpExpect, alice.id.String(), txID)
+	if status != "success" {
+		t.Error(message)
+	}
+
+	docIdentifier = getDocumentIdentifier(t, res)
+
+	params := map[string]interface{}{
+		"document_id": docIdentifier,
+		"currency":    "USD",
+	}
+	getDocumentAndCheck(t, alice.httpExpect, alice.id.String(), typeInvoice, params, true)
+	getDocumentAndCheck(t, bob.httpExpect, bob.id.String(), typeInvoice, params, true)
+	fmt.Println("Host test success")
+
+	// alice adds a funding and shares with charlie
+	res = createFunding(alice.httpExpect, alice.id.String(), docIdentifier, http.StatusOK, defaultFundingPayload([]string{charlie.id.String()}))
+	txID = getTransactionID(t, res)
+	status, message = getTransactionStatusAndMessage(alice.httpExpect, alice.id.String(), txID)
+	if status != "success" {
+		t.Error(message)
+	}
+
+	fundingId = getFundingId(t, res)
+	params = map[string]interface{}{
+		"document_id": docIdentifier,
+		"currency":    "USD",
+		"amount":      "20000",
+		"apr":         "0.33",
+	}
+
+	time.Sleep(2000)
```

Comment:
we shouldn't need to sleep. `getTransactionStatusAndMessage` should have exited after the document consensus.